src/core/Modal/Modal.tsx

```diff
@@ -109,9 +110,11 @@ export const Modal = (props: ModalProps) => {
 
   const originalBodyOverflow = React.useRef('');
 
+  const childrenRef = React.useRef<HTMLDivElement>(null);
+
   // Give focus to overlay for key handling to work.
-  React.useEffect(() => {
-    if (isOpen) {
+  React.useLayoutEffect(() => {
+    if (isOpen && !overlayRef.current?.contains(document.activeElement)) {
       overlayRef.current?.focus();
```

Comment:
When modal is opened using keyboard, there is a visible outline around the overlay. So we should focus the dialog instead, and update the focus styling for the dialog.

Reply:
Amm.. idk about this one.. Not sure, which one should be focused (we can adjust styling of both, of course)

Reply:
how would you style focus on overlay?